Firestore/core/test/firebase/firestore/util/grpc_stream_tester.h

```diff
@@ -20,7 +20,10 @@
 #include <initializer_list>
```

Comment:
@wilhuff General question: with each refactoring, I find this class to resemble `Datastore` more and more. Do you think it would be worthwhile deliberately trying to bring it closer (perhaps make it `FakeDatastore` or something)?

Reply:
I don't have any strong feelings about this--do whatever makes sense.

One thing to note, though: FakeFoo or MockFoo are usually test doubles for Foo, used by tests of things that otherwise use Foo. The GrpcStreamTester seems to work the other way: it's a means of testing pieces that fit into Datastore, which isn't the same thing.

It seems unlikely that a single object will be able to perform both roles without becoming as complex as the thing it's replacing so if I'm understanding your intent, I think this may not be a good path.